app/models/host/base.rb

```diff
@@ -112,7 +112,7 @@ def create_new_host_when_facts_are_uploaded?
 
     # expect a facts hash
     def import_facts(facts)
-      return false unless create_new_host_when_facts_are_uploaded?
+      return false if !create_new_host_when_facts_are_uploaded? && new_record?
```

Comment:
nitpick: new_record? is probably better to check first, as it doesn't need to query another object (such as settings).